File: src/api/system/systemSetting.ts
import { post } from '@/utils/request'

export function getLogoConfig<T>() {
  return post<T>({
    url: '/systemSetting/logo/get',
  })
}

export function setLogoConfig<T>(data: { imageUrl: string; size: number; useCDN?: boolean; cdnUrl?: string }) {
  return post<T>({
    url: '/systemSetting/logo/set',
    data,
  })
}

export function uploadLogo<T>(data: FormData) {
  return post<T>({
    url: '/systemSetting/logo/upload',
    data,
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export function getBackgroundConfig<T>() {
  return post<T>({
    url: '/systemSetting/background/get',
  })
}

export function setBackgroundConfig<T>(data: { imageUrl: string; displayMode: string; useCustomUrl?: boolean; customUrl?: string }) {
  return post<T>({
    url: '/systemSetting/background/set',
    data,
  })
}

export function uploadBackground<T>(data: FormData) {
  return post<T>({
    url: '/systemSetting/background/upload',
    data,
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export function getPresetBackgrounds<T>() {
  return post<T>({
    url: '/systemSetting/background/preset',
  })
}

File: service/router/system/systemSetting.go
package system

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitSystemSettingRouter(router *gin.RouterGroup) {
	api := api_v1.ApiGroupApp.ApiSystem.SystemSettingApi
	
	// 公开读取（无需登录，供登录页/首页展示）
	rPublic := router.Group("", middleware.PublicModeInterceptor)
	{
		rPublic.POST("/systemSetting/logo/get", api.GetLogoConfig)
		rPublic.POST("/systemSetting/background/get", api.GetBackgroundConfig)
		rPublic.POST("/systemSetting/background/preset", api.GetPresetBackgrounds)
	}

	// 管理端读写（按细粒度权限校验）
	r := router.Group("", middleware.LoginInterceptor)
	{
		r.POST("/systemSetting/logo/set", middleware.PermissionInterceptor("setting:edit"), api.SetLogoConfig)
		r.POST("/systemSetting/logo/upload", middleware.PermissionInterceptor("setting:edit"), api.UploadLogo)
		r.POST("/systemSetting/background/set", middleware.PermissionInterceptor("setting:edit"), api.SetBackgroundConfig)
		r.POST("/systemSetting/background/upload", middleware.PermissionInterceptor("setting:edit"), api.UploadBackground)
	}
}